gpiozero/boards.py

```diff
@@ -178,9 +200,96 @@ def _blink_device(self, on_time, off_time, fade_in_time, fade_out_time, n, fps=5
                 break
 
 
+class LEDBarGraph(LEDCollection):
+    """
+    Extends :class:`CompositeDevice` to control a line of LEDs representing a
+    bar graph. Positive values (0 to 1) light the LEDs from first to last.
+    Negative values (-1 to 0) light the LEDs from last to first.
+
+    The following example turns on all the LEDs on a board containing 5 LEDs
+    attached to GPIO pins 2 through 6::
+
+        from gpiozero import LEDBarGraph
+
+        graph = LEDBarGraph(2, 3, 4, 5, 6)
+        graph.value = 2/5  # Light the first two LEDs only
+        graph.value = -2/5 # Light the last two LEDs only
+        graph.off()
```

Comment:
If it was one of the recipes at the front, yeah - I'm a little less bothered about examples in the API docs (want to keep them short and sweet partly so they're easy to read at the command line; if they demonstrate the principle I figure that's enough).